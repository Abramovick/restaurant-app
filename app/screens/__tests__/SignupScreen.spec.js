import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { shallow, } from 'enzyme';
import ConnectedSignup, { SignupPage } from '../SignupScreen';

jest.mock('redux-form/lib/Field', () => 'Field');

// Test objects

const items = [{
  name: 'mariam',
  ImageUrl: 'gdfgds',
  Price: 700,
  description: 'the hello is in the world '
},
{
  name: 'tope',
  ImageUrl: 'gdfgds',
  Price: 700,
  description: 'the hello is in the world'
}];

const navigation = {
  navigate: jest.fn(),
  state: {
    params: {
      name: 'mariam',
      ImageUrl: 'gdfgds',
      Price: 700,
      description: 'the hello is in the world '
    }
  }
};

const response = {
  data: {
    signupUser: {
      token: 12345678,
      firstName: 'mariam'
    }
  }
};

const values = {
  firstName: 'mariam',
  lastName: 'tonia',
  phoneNum: '432524523423',
  email: 'mariam3105@gmail.com',
  password: 'rw840424'
};
// Mocked functions
const handleSubmit = jest.fn();
const rejectedSignupMutation = () => Promise.reject(new Error('not found'));
const resolvedsignupMutation = () => response;

// Test suites
describe('PRODUCTSCREEN --- Snapshot', () => {
  it('renders the links screen here', async () => {
    const tree = renderer.create(<SignupPage
      navigation={navigation}
      handleSubmit={handleSubmit}
      signupUserMutation={rejectedSignupMutation}
      />);
    expect(tree.getInstance().handlePress(values)).toMatchSnapshot();
    // expect(SignupPage.navigationOptions(navigation)).toMatchSnapshot();
    expect(tree).toMatchSnapshot();
  });

  it('renders the links screen here', async () => {
    const tree = renderer.create(<SignupPage
      navigation={navigation}
      handleSubmit={handleSubmit}
      signupUserMutation={resolvedsignupMutation}
      />);
    expect(tree.getInstance().handlePress(values)).toMatchSnapshot();
    // expect(SignupPage.navigationOptions(navigation)).toMatchSnapshot();
    expect(tree).toMatchSnapshot();
  });
});

describe('HOMESCREEN --- Shallow rendering + passing the store directly', () => {
  const initialState = {
    cart: {
      items
    },
    user: {
      loginState: true
    }
  };
  const mockStore = configureMockStore();
  let store, container;

  beforeEach(() => {
    store = mockStore(initialState);
    container = shallow(<ConnectedSignup store={store}/>);
  });

  it('renders the connected component', () => {
    expect(container.length).toEqual(1);
  });
});
