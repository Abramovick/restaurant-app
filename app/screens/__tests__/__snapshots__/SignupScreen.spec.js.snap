// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PRODUCTSCREEN --- Snapshot renders the links screen here 1`] = `
Promise {
  "_40": 0,
  "_55": null,
  "_65": 0,
  "_72": null,
}
`;

exports[`PRODUCTSCREEN --- Snapshot renders the links screen here 2`] = `
<View
  style={
    Object {
      "backgroundColor": "#D57E56",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "paddingTop": 30,
      }
    }
    style={
      Object {
        "backgroundColor": "#D57E56",
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "marginHorizontal": 30,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
        <Field
          component={[Function]}
          name="firstName"
          placeholder="First Name"
        />
        <Field
          component={[Function]}
          name="lastName"
          placeholder="Last Name"
        />
        <Field
          component={[Function]}
          name="email"
          placeholder="E-mail"
        />
        <Field
          component={[Function]}
          name="password"
          placeholder="Password"
          secureTextEntry={true}
        />
        <Field
          component={[Function]}
          name="passwordConfirm"
          placeholder="Confirm Password"
          secureTextEntry={true}
        />
        <Field
          component={[Function]}
          name="phoneNum"
          placeholder="Phone Number"
        />
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "backgroundColor": "#F7C04C",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "elevation": 2,
              "flexDirection": "row",
              "height": 40,
              "justifyContent": "center",
              "marginTop": "4%",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontSize": 12,
              }
            }
          >
                Submit
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`PRODUCTSCREEN --- Snapshot renders the links screen here 3`] = `
Promise {
  "_40": 0,
  "_55": null,
  "_65": 0,
  "_72": null,
}
`;

exports[`PRODUCTSCREEN --- Snapshot renders the links screen here 4`] = `
<View
  style={
    Object {
      "backgroundColor": "#D57E56",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "paddingTop": 30,
      }
    }
    style={
      Object {
        "backgroundColor": "#D57E56",
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "marginHorizontal": 30,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "color": "white",
                  "fontSize": 60,
                },
              ],
              Object {
                "fontFamily": "ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
          virtual={undefined}
        >
          
        </Text>
        <Field
          component={[Function]}
          name="firstName"
          placeholder="First Name"
        />
        <Field
          component={[Function]}
          name="lastName"
          placeholder="Last Name"
        />
        <Field
          component={[Function]}
          name="email"
          placeholder="E-mail"
        />
        <Field
          component={[Function]}
          name="password"
          placeholder="Password"
          secureTextEntry={true}
        />
        <Field
          component={[Function]}
          name="passwordConfirm"
          placeholder="Confirm Password"
          secureTextEntry={true}
        />
        <Field
          component={[Function]}
          name="phoneNum"
          placeholder="Phone Number"
        />
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "backgroundColor": "#F7C04C",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "elevation": 2,
              "flexDirection": "row",
              "height": 40,
              "justifyContent": "center",
              "marginTop": "4%",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#fff",
                    "fontSize": 24,
                    "marginLeft": 16,
                    "marginRight": 0,
                    "paddingTop": 2,
                  },
                ],
                Object {
                  "fontFamily": "ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
            virtual={undefined}
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontSize": 12,
              }
            }
          >
                Submit
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
